src/nvim/generators/gen_api_dispatch.lua

```diff
@@ -33,6 +33,19 @@ local function_names = {}
 
 local c_grammar = require('generators.c_grammar')
 
+-- C code snippets wraping actual api call
+-- /*** API CALL ***/ gets replaced with actual call
+local function_wraps = {
+  nvim_exec = [[
+
+    const sctx_T save_current_sctx = current_sctx;
+    current_sctx.sc_sid = SID_LUA;
```

Comment:
hmm, can't we do this around when calling _into_ lua? then we don't need this hack.

Reply:
I don't know how to tell if api is called from lua or over rpc . That would definatly be prefared way .

Reply:
I mean when non-lua code is calling lua code (finite list of cases in executor.c)

Reply:
Here lua code isn't being called . lua code is calling viml .

Reply:
I did not restate what the code already does. I gave a suggestion how the code could be simplified.

Reply:
I understood what you meant later later . And tried changing current_sctx in lua_pcall since but that added good amount of overhead to all lua stuff . So went with channel_id . I think using channel_id is a good solution to this .